### PR TITLE
feat: support 1D, 2D, and 3D avg and max pooling dynamo converters

### LINKED ISSUE #2200: Exposing IPoolingLayer in dynamo.conversion.impl

### PR DESCRIPTION
**Pull request by zewenli98**
# Description

Support 1D, 2D, and 3D avg and max pooling dynamo converters.

Fixes #2200 

## Type of change

- New feature (non-breaking change which adds functionality)

# Checklist:

- [x] My code follows the style guidelines of this project (You can use the linters)
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas and hacks
- [ ] I have made corresponding changes to the documentation
- [x] I have added tests to verify my fix or my feature
- [x] New and existing unit tests pass locally with my changes
- [x] I have added the relevant labels to my PR in so that relevant reviewers are notified

### CODE REVIEW

**Review by apbose — APPROVED** [review]

File: py/torch_tensorrt/dynamo/conversion/impl/pool.py [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+    if divisor_override is not None:
+        raise RuntimeError("divisor_override is not yet supported!")
+
+    dim = len(kernel_size)
+
+    kernel_size = extend_attr_to_tuple(kernel_size, dim)
+
+    if stride is None:
+        stride = kernel_size
+    else:
```
Comment (gs-olive) [inline py/torch_tensorrt/dynamo/conversion/impl/pool.py:43]: By the function documentation and the [schema](https://github.com/pytorch/pytorch/blob/6aae636f69abebfb8cffea9f7694f9d87929faf9/aten/src/ATen/native/native_functions.yaml#L3736), it should not be allowed for stride to be `None`, so this can be removed

File: py/torch_tensorrt/dynamo/conversion/impl/pool.py [hunk 2]
```diff
@@ -0,0 +1,106 @@
+from typing import Optional, Sequence, Union
+
+# @manual=//deeplearning/trt/python:py_tensorrt
```
Comment (gs-olive) [inline py/torch_tensorrt/dynamo/conversion/impl/pool.py:3]: This can be removed

File: py/torch_tensorrt/dynamo/conversion/impl/pool.py [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+import tensorrt as trt
+from torch.fx.node import Target
+from torch_tensorrt.dynamo.conversion.converter_utils import extend_attr_to_tuple
+from torch_tensorrt.fx.converters.converter_utils import (
+    SourceIR,
```
Comment (gs-olive) [inline py/torch_tensorrt/dynamo/conversion/impl/pool.py:8]: Switch to using the `torch_tensorrt.dynamo` version of `SourceIR`

File: py/torch_tensorrt/dynamo/conversion/impl/pool.py [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+    if ceil_mode is not False:
+        raise RuntimeError("ceil_mode is not yet supported!")
+
+    dim = len(kernel_size)
+
+    kernel_size = extend_attr_to_tuple(kernel_size, dim)
+
+    if stride is None:
+        stride = kernel_size
+    else:
```
Comment (gs-olive) [inline py/torch_tensorrt/dynamo/conversion/impl/pool.py:90]: See above comment (can remove)

**Review by gs-olive — COMMENTED** [review]
Left a few code comments, otherwise looks good to me

File: py/torch_tensorrt/dynamo/conversion/aten_ops_converters.py [hunk 5]
```diff
@@ -1378,3 +1378,88 @@ def aten_ops_linear(
...
+        _LOGGER.debug(
+            f"Currently we don't support specifying ceil_mode, got ceil_mode={ceil_mode}."
+        )
+        return False
+
+    return True
+
+
+# Note: MaxPool1d uses max_pool2d as it converts to 2D first.
+@dynamo_tensorrt_converter(torch.ops.aten.max_pool2d.default, capability_validator=max_pool_param_validator)  # type: ignore[misc]
```
Comment (gs-olive) [inline py/torch_tensorrt/dynamo/conversion/aten_ops_converters.py:1445]: Could `max_pool1d` support be added here as well? [Schema](https://github.com/pytorch/pytorch/blob/bf7307adf8941960adbba2abb77f04ba23bed692/aten/src/ATen/native/native_functions.yaml#L3734)

File: py/torch_tensorrt/dynamo/conversion/impl/pool.py (hunk 1 above)
Comment (zewenli98) [reply py/torch_tensorrt/dynamo/conversion/impl/pool.py:43]: Thanks for pointing it out! According to the schema, the stride could be `[]` by default, so I made changes accordingly.

File: py/torch_tensorrt/dynamo/conversion/aten_ops_converters.py (hunk 5 above)
Comment (zewenli98) [reply py/torch_tensorrt/dynamo/conversion/aten_ops_converters.py:1445]: I can add `torch.ops.aten.max_pool1d.default` but it won't be used. Even for `torch.nn.AvgPool1d`, it still calls `torch.ops.aten.avg_pool2d.default`, as you can see in the test file: https://github.com/pytorch/TensorRT/pull/2317/files#diff-9fce39bc42c66d2866c41665779cab7da0a4d3fe54576925e2b66c17a1cf1ebfR42
But anyways, the 1d schema looks same as others, so I added here.

Comment (gs-olive) [reply py/torch_tensorrt/dynamo/conversion/aten_ops_converters.py:1447]: Thanks for that - I plan to add a lowering pass which will lead us to that converter, so it will still be helpful.

**Review by gs-olive — APPROVED** [review]
Looks good to me, pending CI!
